apps/emqx_authn/src/simple_authn/emqx_authn_mnesia.erl

```diff
@@ -91,6 +90,7 @@ fields(config) ->
     [ {mechanism,               {enum, ['password-based']}}
     , {backend,                 {enum, ['built-in-database']}}
     , {user_id_type,            fun user_id_type/1}
+    , {user_group,              fun user_group/1}
```

Comment:
Here is an important change concerning `user_group`.

I moved it to the config part because it seems to be related to the persistent storage of users.
It seems unlikely that a user would like
* To have inevitably different users for each of his chains.
* To have all of their users _destroyed_ when they remove _authenticator config_, e.g. through API.

More likely, a user would like to have their users persistent, like with other DB providers, when destroying config destroys only DB connection.

So, `user_group` moved to config now serves more like a _storage address_, like `host:port/database`.

I may be wrong with this :)

Reply:
Sounds like a reasonable argument to me. :slightly_smiling_face:

Reply:
@savonarola Hi. In my imagination, when users create multiple authentication instances that use the built-in database, they don’t need to care about how to isolate data between multiple instances.

Now `user_group` has become a field that users must understand its usefulness, which adds complexity to a certain extent.

At the same time, we seem to lack the ability to allow users to manage these data, and how we can ensure that users will clean up these data in a timely manner.

I think the latter will be the key point.

Reply:
bq. At the same time, we seem to lack the ability to allow users to manage these data, and how we can ensure that users will clean up these data in a timely manner.

Considering the introduced change, I supposed that users would be able to manage users explicitly: adding and deleting users through API.

At the same time, they wouldn't have their users (unexpectedly?) removed when they, for example, try to use an alternative backend and temporarily (as they may think) drop mnesia authentication section with API.

Reply:
@tigercl if we want to isolate users by default, we may use authenticator id as default `group_id` (like it was previously done).